APE_Black_Formatting.rst

```diff
@@ -0,0 +1,286 @@
+.. _Black: https://github.com/psf/black
+.. _Django DEP-0008: https://github.com/django/deps/blob/main/final/0008-black.rst
```

Comment:
I feel bringing Django into the argument is comparing apples to oranges. black is absolutely awful with mathematical expression formatting as well as with nested containers, so comparing the astropy case to e.g. scipy would be more realistic.

Reply:
Django is similar to astropy in being a large community-driven open-source project with a range of experience levels for contributors. So the comparison is useful in terms of developer and community experience.

As mentioned elsewhere, formatting of the code and the small fraction of cases where this goes badly is not a problem when you actually run black on astropy and look at the results. For instance try `modeling`, which is probably one of the more math-heavy parts of astropy. A lot of equations are changed, and IMO the changes are mostly improvements and at worst neutral.

But again, if the developer and PR reviewer both agree that black is doing badly then the that code can be skipped.